addons/website_crm_score/models/crm_lead.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+from openerp.http import request
+from openerp import api, fields, models, SUPERUSER_ID
+import md5
+
+
+class Lead(models.Model):
+    _inherit = 'crm.lead'
+
+    @api.one
+    def _count_pageviews(self):
+        self.pageviews_count = len(self.score_pageview_ids)
+
+    @api.depends('score_ids', 'score_ids.value')
+    def _compute_score(self):
+        self._cr.execute("""
+             SELECT
+                lead_id, COALESCE(sum(s.value), 0) as sum
+             FROM
+                crm_lead_score_rel rel
+             LEFT JOIN
+                website_crm_score s ON (s.id = rel.score_id)
+             WHERE lead_id = any(%s)
+             GROUP BY lead_id
+             """, (self.ids,))
+        scores = dict(self._cr.fetchall())
+        for lead in self:
+            lead.score = scores.get(lead.id, 0)
+
+    score = fields.Float(compute='_compute_score', store=True)
+    score_ids = fields.Many2many('website.crm.score', 'crm_lead_score_rel', 'lead_id', 'score_id', string='Score')
+    score_pageview_ids = fields.One2many('website.crm.pageview', 'lead_id', string='Page Views', help="List of (tracked) pages seen by the owner of this lead")
+    assign_date = fields.Datetime(string='Assign Date', help="Date when the lead has been assigned via the auto-assignation mechanism")
+    pageviews_count = fields.Integer('# Page Views', compute='_count_pageviews')
+    lang_id = fields.Many2one('res.lang', string='Language', help="Language from the website when lead has been created")
+
+    def encode(self, lead_id):
+        encrypted_lead_id = md5.new("%s%s" % (lead_id, self.get_key())).hexdigest()
```

Comment:
variable name is wrong (that's a MAC not an encryption), and md5 is not designed for MACs, this should use hmac.